src/color.rs

```diff
@@ -1,54 +1,64 @@
 use num_traits::{NumCast, Zero};
-use std::mem;
 use std::ops::{Index, IndexMut};
 
 use buffer::Pixel;
 use traits::Primitive;
 
-/// An enumeration over supported color types and their bit depths
+/// An enumeration over supported color types and bit depths
 #[derive(Copy, PartialEq, Eq, Debug, Clone, Hash)]
 pub enum ColorType {
-    /// Pixel is grayscale
-    Gray(u8),
-
-    /// Pixel contains R, G and B channels
-    RGB(u8),
-
     /// Pixel is an index into a color palette
     Palette(u8),
 
-    /// Pixel is grayscale with an alpha channel
-    GrayA(u8),
-
-    /// Pixel is RGB with an alpha channel
-    RGBA(u8),
-
-    /// Pixel contains B, G and R channels
-    BGR(u8),
-
-    /// Pixel is BGR with an alpha channel
-    BGRA(u8),
-
+    /// Pixel is 1-bit luminance
+    L1,
+
+    /// Pixel is 8-bit luminance
+    L8,
+    /// Pixel is 8-bit luminance with an alpha channel
+    LA,
+    /// Pixel contains 8-bit R, G and B channels
+    RGB,
+    /// Pixel is 8-bit RGB with an alpha channel
+    RGBA,
+
+    /// Pixel is 16-bit luminance
+    L16,
+    /// Pixel is 16-bit luminance with an alpha channel
+    LA16,
+    /// Pixel is 16-bit RGB
+    RGB16,
+    /// Pixel is 16-bit RGBA
+    RGBA16,
+
+    /// Pixel contains 8-bit B, G and R channels
+    BGR,
+    /// Pixel is 8-bit BGR with an alpha channel
+    BGRA,
 }
 
 /// Returns the number of bits contained in a pixel of `ColorType` ```c```
 pub fn bits_per_pixel(c: ColorType) -> usize {
     match c {
-        ColorType::Gray(n) => n as usize,
-        ColorType::GrayA(n) => 2 * n as usize,
-        ColorType::RGB(n) | ColorType::Palette(n)| ColorType::BGR(n) => 3 * n as usize,
-        ColorType::RGBA(n) | ColorType::BGRA(n) => 4 * n as usize,
+        ColorType::L1 => 1,
+        ColorType::L8 => 8,
+        ColorType::L16 => 16,
+        ColorType::Palette(n) => 3 * n as usize,
+        ColorType::LA => 16,
+        ColorType::RGB | ColorType::BGR => 24,
+        ColorType::RGBA | ColorType::BGRA | ColorType::LA16 => 32,
+        ColorType::RGB16 => 48,
+        ColorType::RGBA16 => 64,
     }
 }
 
 /// Returns the number of color channels that make up this pixel
 pub fn num_components(c: ColorType) -> usize {
     match c {
-        ColorType::Gray(_) => 1,
-        ColorType::GrayA(_) => 2,
-        ColorType::RGB(_) | ColorType::Palette(_) | ColorType::BGR(_)=> 3,
```

Comment:
Seems like Palette should have only a single component, but didn't want to break this without asking

Reply:
Mhm, yes, seems mostly reasonable if it is opaque anyways. But I'm not quite sure myself.